### PR TITLE
refactor(client): migrate link to Button

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR:
- Migrates some Gatsby `Link` to `Button`
- Removes the `.btn-primary` overrides
- Is related to #52030

### CODE REVIEW

File: client/src/templates/Introduction/components/challenges.tsx [hunk 1]
```diff
@@ -66,15 +68,12 @@ function Challenges({
     <>
       {firstIncompleteChallenge && (
         <div className='challenge-jump-link'>
-          <Link
-            className='btn btn-primary'
-            to={firstIncompleteChallenge.fields.slug}
-          >
+          <Button size='small' href={firstIncompleteChallenge.fields.slug}>
```
[the remark this answers is not in the document]
Comment (moT01) [reply client/src/templates/Introduction/components/challenges.tsx:71]: This looks a little smaller than the old button.

File: client/src/components/profile/components/certifications.tsx [hunk 2]
```diff
@@ -46,15 +47,16 @@ function CertButton({ username, cert }: CertButtonProps): JSX.Element {
   const { t } = useTranslation();
   return (
     <>
-      <Link
-        className='btn btn-lg btn-primary btn-block'
-        to={`/certification/${username}/${cert.certSlug}`}
+      <Button
```
[the remark this answers is not in the document]
Comment (moT01) [reply client/src/components/profile/components/certifications.tsx:50]: These look a little bigger than the old buttons.

**Review by moT01 — APPROVED** [review]
LGTM 🎉 

I left a few comments about the size of the buttons, but I think it's fine. Was just pointing it out.

File: client/src/components/layouts/global.css [hunk 3]
```diff
@@ -229,35 +229,24 @@ button:hover,
 input[type='submit']:hover,
 .btn:hover,
 .btn-default.active:hover,
-.btn-default.active,
-.btn-primary:hover {
```
[the remark this answers is not in the document]
Comment (ahmaxed) [reply client/src/components/layouts/global.css:233]: btn-default does not exist as well.

**Review by ahmaxed — APPROVED** [review]
Thanks for migrating the buttons and removing classes. As Tom mentioned, there are slight discrepancies. We could address the paddings and standardize the font sizes after migration.

**Pull request merged**